source/mainnet/net/references/grpc.rst

```diff
@@ -0,0 +1,886 @@
+.. _grpc-documentation:
+
+====================
+ gRPC documentation
+====================
+
+The Concordium node has a `gRPC <https://grpc.io/>`_ interface that enables
+queries, sending of transactions, and more.
+While the gRPC interface is powerful, it is not the most convenient tool.
+We provide a number of SDKs that build on top of the gRPC interface, which are
+much more ergonomic to use.
+
+These are the SDKs we currently provide:
+
+- `Concordium Rust SDK <https://github.com/Concordium/concordium-rust-sdk>`_
+- `Concordium Javascript (Node / Web) SDK <https://github.com/Concordium/concordium-node-sdk-js>`_
+- `Concordium Java SDK <https://github.com/Concordium/concordium-java-sdk>`_
+- `Concordium .NET (C#) SDK <https://github.com/Concordium/concordium-net-sdk>`_
+- `Concordium Go SDK <https://github.com/Concordium/concordium-go-sdk>`_
+
+The gRPC interface uses a mixture of protobuf defined types and JSON. For the
+JSON types, a `JSON Schema <https://json-schema.org/>`_ is provided.
+
+For the JSON schemas, you may find these two tools useful:
+
+- `JSON Schema Faker <https://json-schema-faker.js.org/>`_: Generates fake data
+  that conforms to the provided schema.
+- `JSON Schema Validator <https://www.jsonschemavalidator.net/>`_: Validates
+  whether the JSON input follows the provided schema.
+
+
+Notation
+========
+
+This page uses the following notation:
+
+- ``?a`` means ``a`` OR ``null``.
+
+  - JSON schemas for ``?a``-types, will only specify the ``a``. But the node may
+    also return ``null``.
```

Comment:
I don't understand this. Won't it say the field is optional?

Reply:
What I meant was that the schemas only handled `a` and not `?a`, which wasn't a great solution.
So, as we discussed, I've updated the schemas so they now handle `?a` for the relevant endpoints. (I've queried the node via `concordium-client raw` to determine which endpoints might return null.)